core/src/main/java/com/graphhopper/storage/BaseGraph.java

```diff
@@ -822,41 +829,81 @@ private void setWayGeometry_( PointList pillarNodes, long edgePointer, boolean r
                 throw new IllegalArgumentException("Cannot use pointlist which is " + pillarNodes.getDimension()
                         + "D for graph which is " + nodeAccess.getDimension() + "D");
 
+            long existingGeoRef = Helper.toUnsignedLong(edges.getInt(edgePointer + E_GEO));
+            
             int len = pillarNodes.getSize();
             int dim = nodeAccess.getDimension();
-            long tmpRef = nextGeoRef(len * dim);
-            edges.setInt(edgePointer + E_GEO, Helper.toSignedInt(tmpRef));
-            long geoRef = (long) tmpRef * 4;
-            byte[] bytes = new byte[len * dim * 4 + 4];
-            ensureGeometry(geoRef, bytes.length);
-            bitUtil.fromInt(bytes, len, 0);
-            if (reverse)
-                pillarNodes.reverse();
-
-            int tmpOffset = 4;
-            boolean is3D = nodeAccess.is3D();
-            for (int i = 0; i < len; i++)
+            if (existingGeoRef > 0)
             {
-                double lat = pillarNodes.getLatitude(i);
-                bitUtil.fromInt(bytes, Helper.degreeToInt(lat), tmpOffset);
-                tmpOffset += 4;
-                bitUtil.fromInt(bytes, Helper.degreeToInt(pillarNodes.getLongitude(i)), tmpOffset);
-                tmpOffset += 4;
-
-                if (is3D)
-                {
-                    bitUtil.fromInt(bytes, Helper.eleToInt(pillarNodes.getElevation(i)), tmpOffset);
-                    tmpOffset += 4;
-                }
+            	long existingGeoRefPos = existingGeoRef * 4L;
+				final int count = wayGeometry.getInt(existingGeoRefPos);
+            	if (len <= count)
+            	{
+            		// If old array was smaller, erase contents
+            		if (len < count)
+            		{
+            			eraseWayGeometryAtGeoRef(count, existingGeoRef);
```

Comment:
You should be able to just do another commit in your branch and it should show up here. But I'll do the code formatting and merge this with your changes (+could remove the method) - let me know what to do :)

Regarding compressibility: that is indeed true. But I'm unsure how we can apply compression to the geometry and still keep everything else working & fast.